src/main/java/nwoolcan/viewmodel/brewery/warehouse/WarehouseViewModel.java

```diff
@@ -1,141 +1,163 @@
      */
     public int getnIngredientAvailable() {
-        return nIngredientAvailable;
+        return warehouse.getStocks(INGREDIENT_AVAILABLE_QUERY).size();
     }
     /**
      * Return the number of expired {@link nwoolcan.model.brewery.warehouse.stock.BeerStock}.
      * @return the number of expired {@link nwoolcan.model.brewery.warehouse.stock.BeerStock}.
      */
     public int getnBeerExpired() {
-        return nBeerExpired;
+        return warehouse.getStocks(BEER_EXPIRED_QUERY).size();
     }
     /**
      * Return the number of expired {@link nwoolcan.model.brewery.warehouse.stock.Stock} with a misc {@link nwoolcan.model.brewery.warehouse.article.Article}.
      * @return the number of expired {@link nwoolcan.model.brewery.warehouse.stock.Stock} with a misc {@link nwoolcan.model.brewery.warehouse.article.Article}.
      */
     public int getnMiscExpired() {
-        return nMiscExpired;
+        return warehouse.getStocks(MISC_EXPIRED_QUERY).size();
     }
     /**
      * Return the number of used {@link nwoolcan.model.brewery.warehouse.stock.Stock} with an {@link nwoolcan.model.brewery.warehouse.article.IngredientArticle}.
      * @return the number of used {@link nwoolcan.model.brewery.warehouse.stock.Stock} with an {@link nwoolcan.model.brewery.warehouse.article.IngredientArticle}.
      */
     public int getnIngredientExpired() {
-        return nIngredientExpired;
+        return warehouse.getStocks(INGREDIENT_EXPIRED_QUERY).size();
     }
     /**
      * Return the number of used {@link nwoolcan.model.brewery.warehouse.stock.BeerStock}.
      * @return the number of used {@link nwoolcan.model.brewery.warehouse.stock.BeerStock}.
      */
     public int getnBeerUsed() {
-        return nBeerUsed;
+        return warehouse.getStocks(BEER_USED_QUERY).size();
     }
     /**
      * Return the number of used {@link nwoolcan.model.brewery.warehouse.stock.Stock} with a misc {@link nwoolcan.model.brewery.warehouse.article.Article}.
      * @return the number of used {@link nwoolcan.model.brewery.warehouse.stock.Stock} with a misc {@link nwoolcan.model.brewery.warehouse.article.Article}.
      */
     public int getnMiscUsed() {
-        return nMiscUsed;
+        return warehouse.getStocks(MISC_USED_QUERY).size();
     }
     /**
      * Return the number of used {@link nwoolcan.model.brewery.warehouse.stock.Stock} with an {@link nwoolcan.model.brewery.warehouse.article.IngredientArticle}.
      * @return the number of used {@link nwoolcan.model.brewery.warehouse.stock.Stock} with an {@link nwoolcan.model.brewery.warehouse.article.IngredientArticle}.
      */
     public int getnIngredientUsed() {
-        return nIngredientUsed;
+        return warehouse.getStocks(INGREDIENT_USED_QUERY).size();
     }
     /**
      * Returns the list of all {@link AbstractStockViewModel}.
      * @return a list of {@link AbstractStockViewModel}.
      */
     public List<AbstractStockViewModel> getStocks() {
```

Comment:
the purpose of using the query is so you can build your custom one from the view and pass it to the controller that has a method with a parameter of type query

Reply:
sure. In this particular implementation I have to do the queries in order to have a sort of dashboard for the view. There will be the method in the controller in order to do custom queries.

Reply:
> There will be the method in the controller in order to do custom queries.

it is exacly what you want to AVOID.
the view has all infos to build the query and should be build there for general use.
for specific things it is ok to build it from the controller

Reply:
I think i didn't explain well. The query will be built by the view and the query object will be given to the controller

Reply:
ok then. but i dont see any method in the controller that gets a query from parameter.
do you mean you will write it later?

Reply:
sure i'm doing it :)

Reply:
it will also return a view stocks instead of truestocks

Reply:
> sure i'm doing it :)

i mean, you can just do it, it is an exact copy of the below but with the query passed by parameter

Reply:
I will do it only for custom queries. I don't want the reference to the controller in the viewmodel object

Reply:
Since this involves the Controller, we can close this PR and apply this modification in another PR.